app/controllers/hiring_staff/vacancies/schools_controller.rb

```diff
@@ -0,0 +1,85 @@
+class HiringStaff::Vacancies::SchoolsController < HiringStaff::Vacancies::ApplicationController
+  include OrganisationHelper
+
+  before_action :verify_school_group
+  before_action :set_up_url
+  before_action :set_up_previous_step_path
+  before_action :set_multiple_schools
+  before_action :set_organisation_options
+  before_action only: %i[create update] do
+    set_up_form(SchoolsForm)
+  end
+
+  def show
+    attributes = @vacancy.present? ? @vacancy.attributes : session[:vacancy_attributes]
+    return redirect_to next_step if job_location == 'central_office'
+    @form = SchoolsForm.new(attributes.symbolize_keys)
+  end
+
+  def create
+    @form.vacancy.readable_job_location = readable_job_location(job_location, school_name: school&.name,
+                                                                schools_count: @form.organisation_ids&.count)
+    store_vacancy_attributes(@form.vacancy.attributes)
+    session[:vacancy_attributes]['organisation_id'] = @form.organisation_id
+    session[:vacancy_attributes]['organisation_ids'] = @form.organisation_ids
```

Comment:
Is there some reason we can't use the same attribute for both cases? A single-element organisation_ids array?

Would save a lot of pieces of logic here and there.

Reply:
While it might save logic in some places, it would introduce logic in lots of other places too because the fields on the form are distinctly different. I'd prefer to just have one form input and users select 1 or more schools, but here we are 🤷‍♂️